app/src/js/components/FormSchema/schema.js

```diff
@@ -36,65 +48,91 @@ export const traverseSchema = function (schema, fn, path) {
-      set(readOnlyRemoved, accessor, get(data, accessor));
+      set(readOnlyRemoved, path, get(data, path));
     }
   });
 
   // filter fields that are not in the schema
   const esFields = ['queriedAt', 'timestamp', 'stats'];
-  const nonSchemaFields = Object.keys(data).filter(f => (
-    schemaFields.indexOf(f) === -1 && esFields.indexOf(f) === -1)
+  const nonSchemaFields = Object.keys(data).filter(
+    (f) => !schemaFields.includes(f) && !esFields.includes(f)
   );
 
   // add them to the list of fields
-  nonSchemaFields.forEach(f => set(readOnlyRemoved, f, get(data, f)));
+  nonSchemaFields.forEach((f) => set(readOnlyRemoved, f, get(data, f)));
 
   return readOnlyRemoved;
 };
 
 // recursively scan a schema object and create a form config from it.
 // returns a flattened representation of the schema.
-export const createFormConfig = function (data, schema, include) {
+export const createFormConfig = function (
+  data,
+  schema,
+  include,
+  exclude,
+  enums = {}
+) {
   data = data || {};
   const fields = [];
-  traverseSchema(schema, function (property, meta, schemaProperty, path) {
-    // If a field isn't user-editable, hide it from the form
-    if (meta.readonly) { return; }
+  const toRegExps = (stringsOrRegExps) =>
+    stringsOrRegExps.map((strOrRE) =>
+      typeof strOrRE === 'string' ? new RegExp(`^${strOrRE}$`, 'i') : strOrRE
+    );
+  const inclusions = toRegExps(include);
+  const exclusions = toRegExps(exclude);
+  const matches = (string) => (regexp) => regexp.test(string);
+  const includeProperty = (path) =>
+    inclusions.some(matches(path)) && !exclusions.some(matches(path));
 
-    // create an object-path-ready accessor string
-    const accessor = path ? path + '.' + property : property;
+  traverseSchema(schema, enums, (path, meta, schemaProperty) => {
+    const fullyQualifiedProperty = path.join('.');
+
+    // If a field isn't user-editable, hide it from the form
+    if (meta.readonly) return;
 
     // if there are included properties, only create forms for those
-    if (Array.isArray(include) && include.indexOf(accessor) === -1) { return; }
+    if (!includeProperty(fullyQualifiedProperty)) return;
 
     // determine the label
-    const required = Array.isArray(schemaProperty.required) &&
-      schemaProperty.required.indexOf(property) >= 0;
+    const property = path[path.length - 1];
+    const required =
+      isArray(schemaProperty.required) &&
+      schemaProperty.required.includes(property);
 
+    const labelText = meta.title || property;
     const label = (
       <span>
-        { path ? <span className='label__path'>{path + ' - '}</span> : null }
-        <span className='label__name'>{meta.title || property}</span>
-        { required ? <span className='label__required'> *</span> : null }
-        { meta.description ? <span className='label__description'> ({meta.description})</span> : null }
+        <span className="label__name">{labelText}</span>
```

Comment:
While we're in here, any chance we can call `displayCase` on the `labelText` to get consistent capitalization?